### PR TITLE
Added upload and periods

### PR DESCRIPTION
**Pull request by jhc-deloitte**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 1]
```diff
@@ -687,6 +738,13 @@ const ParentalLeaveTemplate: ApplicationTemplate<
           },
           lifecycle: pruneAfterDays(970),
           progress: 1,
+          onExit: defineTemplateApi({
+            action:
+              ApiModuleActions.validateApplication &&
+              ApiModuleActions.setBirthDate,
```
Comment (karenbjorg) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:743]: I don't think we will get the birth date when vmst approves the application.

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 2]
```diff
@@ -1455,17 +1516,28 @@ const ParentalLeaveTemplate: ApplicationTemplate<
...
-      setHasApplied: assign((context) => {
+      setDateOfBirth: assign((context) => {
         const { application } = context
         const { answers } = application
-        if (
-          answers.previousState === 'approved' &&
-          application.state === 'vinnumalastofnunApproveEdits'
-        ) {
-          set(answers, 'hasAppliedFor', 'residenceGrant')
+        const data = (application.externalData.dateOfBirth.data as unknown) as {
```
Comment (karenbjorg) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:1521]: I think you should add this to the getApplicationExternalData in parentalLeaveUtils, something like this:
const dateOfBirth = getValueViaPath(externalData, 'dateOfBirth') as string

File: libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts [hunk 3]
```diff
@@ -189,6 +189,27 @@ export class ParentalLeaveService extends BaseTemplateApiService {
     }
   }
 
+  async setBirthDate({ application }: TemplateApiModuleActionProps) {
+    const { applicationFundId } = getApplicationExternalData(
+      application.externalData,
+    )
+    if (applicationFundId) {
```
Comment (karenbjorg) [inline libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts:196]: If applicationFundId is found then we get the birth date?

File: libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts [hunk 4]
```diff
@@ -362,7 +383,23 @@ export class ParentalLeaveService extends BaseTemplateApiService {
     const { applicationFundId } = getApplicationExternalData(
       application.externalData,
     )
-
+    const attachmentsFiles = (application.answers.fileUpload as unknown) as {
```
Comment (karenbjorg) [inline libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts:386]: Can you add this to getApplicationAnswers in parentalLeaveUtils?

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 5]
```diff
@@ -1455,17 +1516,28 @@ const ParentalLeaveTemplate: ApplicationTemplate<
...
+          dateOfBirth: string
+        }
+        if (data.dateOfBirth) {
+          const dateOfBirth = data.dateOfBirth
+          set(answers, 'dateOfBirth', dateOfBirth)
         }
         return context
       }),
+      checkBirthDate: assign((context) => {
```
[the remark this answers is not in the document]
Comment (jhc-deloitte) [reply libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts]: I removed it

File: libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts (hunk 4 above)
Comment (jhc-deloitte) [reply libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts:386]: Yes def.

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts (hunk 1 above)
Comment (jhc-deloitte) [reply libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:743]: No, but we check for it when we are in the approved state, and click the link to residence grant

File: libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts (hunk 3 above)
Comment (jhc-deloitte) [reply libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts:196]: I think this was a brain fart. Should not be there.

**Pull request merged**